### ISSUE TITLE
EfficientNet implementation

### ISSUE THREAD

**Issue by leonid-pishchulin** [dmlc/gluon-cv#796]
Hey, are there any EfficientNet (https://arxiv.org/abs/1905.11946) implementations available? If not, is somebody working on it?

**Comment by KellenSunderland** [comment]
Just for some background: we're mostly interested so no one duplicates effort.

**Comment by zhreshold** [comment]
here's one candidate @sufeidechabei 
To summarize, it might be easy to write the definitions(https://github.com/mnikitin/EfficientNet/blob/master/efficientnet_model.py), but training part is unpredictable yet due to the missing training hyper-parameters in paper. 

Should we split the risk by training the network simultanously?

**Comment by bermanmaxim** [comment]
Note that https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet has the training script that seems to have been used and provides hyperparameters.

**Comment by bermanmaxim** [comment]
Also there might be some differences with the paper, author says "source code is correct": see https://github.com/tensorflow/tpu/issues/383, https://github.com/tensorflow/tpu/issues/390

**Comment by ryanjay0** [comment]
They never said "source code is correct" about tpu issue #390. Did they? Seems like a much larger discrepancy than the padding issues in #383

**Comment by bermanmaxim** [comment]
True, what I meant is that since the author said source code is correct I was assuming the code is what they actually used including for the other issue. But you are right that this resolution discrepancy is a big difference 🤔

**Comment by bermanmaxim** [comment]
Regarding the training of efficientnet, see the remarks of Ross Wightman here: https://forums.fast.ai/t/efficientnet/46978/67 ; it might be that keeping an exponential moving average of the weights during training, for use at testing, helps this family of models a lot.

**Comment by bermanmaxim** [comment]
Similar discussion over here: https://github.com/pytorch/vision/issues/980

**Comment by hetong007** [comment]
@sufeidechabei please check your PR with the resources above.
